File: vbox-app/src/com/kedzie/vbox/api/IPerformanceMetric.java
package com.kedzie.vbox.api;

import java.util.HashMap;
import java.util.Map;

import android.os.Parcel;
import android.os.Parcelable;

import com.kedzie.vbox.soap.KSOAP;
import com.kedzie.vbox.soap.VBoxSvc;

@KSOAP(cacheable=true)  
public interface IPerformanceMetric extends IManagedObjectRef, Parcelable {
    
    static final ClassLoader LOADER = IPerformanceMetric.class.getClassLoader();

    public static Parcelable.Creator<IPerformanceMetric> CREATOR = new Parcelable.Creator<IPerformanceMetric>() {
        @Override
        public IPerformanceMetric createFromParcel(Parcel in) {
            VBoxSvc vmgr =  in.readParcelable(LOADER);
            String id = in.readString();
            Map<String, Object> cache = new HashMap<String, Object>();
            in.readMap(cache, LOADER);
            return (IPerformanceMetric) vmgr.getProxy(IPerformanceMetric.class, id, cache); 
        }

        @Override
        public IPerformanceMetric[] newArray(int size) {
            return new IPerformanceMetric[size];
        }
    };
    
	public String getMetricName();
	public String getDescription();
	public Integer getMinimumValue();
	public Integer getMaximumValue();
	public String getUnit();
	public String getObject();
}

File: vbox-app/src/main/java/com/kedzie/vbox/api/IPerformanceMetric.kt
package com.kedzie.vbox.api

import android.os.Parcelable
import com.kedzie.vbox.soap.Cacheable
import com.kedzie.vbox.soap.Ksoap
import com.kedzie.vbox.soap.KsoapProxy

@KsoapProxy
@Ksoap
interface IPerformanceMetric : IManagedObjectRef, Parcelable {
    @Cacheable("MetricName")
	suspend fun getMetricName(): String
    @Cacheable("Description")
	suspend fun getDescription(): String
    @Cacheable("MinimumValue")
	suspend fun getMinimumValue(): Int
    @Cacheable("MaximumValue")
	suspend fun getMaximumValue(): Int
    @Cacheable("Period")
	suspend fun getPeriod(): Long
    @Cacheable("Count")
	suspend fun getCount(): Long
    @Cacheable("Unit")
	suspend fun getUnit(): String
    @Cacheable("Object")
	suspend fun getObject(): String
}